playbooks/common/openshift-cluster/upgrades/pre/config.yml

```diff
@@ -64,7 +64,9 @@
   - fail: msg="Master running {{ openshift.common.version }} must be upgraded to {{ openshift_version }} before node upgrade can be run."
     when:
     - l_upgrade_nodes_only | default(False) | bool
-    - not openshift.common.version | match(openshift_version)
+    # we may get openshift_version as either 3.9 or 3.9.x
+    - not openshift.common.version.split('.')[0:2] | match(openshift_version.split('.')[0:2])
```

Comment:
I have considered following accurate conditions, but above fix should be enough for version check, I think.

~~~
    - (openshift.common.version.split('.') | length == openshift_version.split('.') | length) and (not openshift.common.version | match(openshift_version))
    - (openshift.common.version.split('.') | length != openshift_version.split('.') | length) and (not openshift.common.version.split('.')[0:2] | match(openshift_version.split('.')[0:2]))
~~~